### DISCUSSION TITLE
Potential Update to dev-requirements.txt

### DISCUSSION THREAD

**Discussion by dcechano** [streamlink/streamlink#5035] [Q&A] [answered]
I recently forked and cloned this repo on to my computer and was setting up my environment. I simply opened the dev-requirements.txt and downloaded every package using pip. 

Then I attempted to run using `python -m streamlink` and got this error:

`  File "C:\Users\super\PycharmProjects\streamlink\src\streamlink\utils\parse.py", line 5, in <module>
    from lxml.etree import HTML, XML
ModuleNotFoundError: No module named 'lxml'
`

Odd. I installed all dev-requirements programmatically. It turns out dev-requirements.txt uses `lxml-stubs ` not `lxml`. NBD, simply run `pip install lxml`. 

Then ran `python -m streamlink` again and got this:

` Traceback (most recent call last):
  File "<frozen runpy>", line 189, in _run_module_as_main
  File "<frozen runpy>", line 148, in _get_module_details
  File "<frozen runpy>", line 112, in _get_module_details
  File "C:\Users\super\PycharmProjects\streamlink\src\streamlink\__init__.py", line 19, in <module>
    from streamlink.api import streams
  File "C:\Users\super\PycharmProjects\streamlink\src\streamlink\api.py", line 1, in <module>
    from streamlink.session import Streamlink
  File "C:\Users\super\PycharmProjects\streamlink\src\streamlink\session.py", line 18, in <module>
    from streamlink.utils.l10n import Localization
  File "C:\Users\super\PycharmProjects\streamlink\src\streamlink\utils\l10n.py", line 5, in <module>
    from pycountry import countries, languages  # type: ignore[import]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'pycountry'
`
Basically I am starting this discussion because I am new to this project and don't want to step on toes by starting an issue. This isn't a "bug" so, does `dev-requirements.txt` need to be updated? What do you guys think?

For your convenience `dev-requirements.txt` is below.
[dev-requirements.txt](https://github.com/streamlink/streamlink/files/10248212/dev-requirements.txt)

**Comment by bastimeyer** [comment, marked as answer]
> Odd. I installed all dev-requirements programmatically.

dev-requirements are for running tests and other stuff. You are completely missing Streamlink's **runtime dependencies**, which are defined by setuptools and which get installed by `python-pip` when you build and install the wheel, preferrably in editable mode.
https://github.com/streamlink/streamlink/blob/9b19e62e448ba8dbe1cb3b3110014fb13bb36898/setup.cfg#L41-L51

Read this.
https://streamlink.github.io/developing.html

**Reply by dcechano** [reply]
I scanned that page before and figured I knew what it was doing already. Shows what I know. Thank you very much sir!

**Reply by gravyboat** [reply]
@dcechano If you're thinking about adding a feature or modifying something please create an issue about it for discussion prior to performing any work just in case it's something we don't want to add you won't waste time.

**Reply by dcechano** [reply]
> @dcechano If you're thinking about adding a feature or modifying something please create an issue about it for discussion prior to performing any work just in case it's something we don't want to add you won't waste time.

@gravyboat I understand sir. I was actually thinking about trying my hand at [Issue #4979](https://github.com/streamlink/streamlink/issues/4979) if that is ok.

**Reply by gravyboat** [reply]
Sounds good!
